### ISSUE TITLE
Skip files that are too large

### ISSUE THREAD

**Issue by eoftedal** [dependency-check/DependencyCheck#148]
If a zip file is larger than Integer.MAX_VALUE and execption is thrown from:

```
ERROR: Processing failed due to a bug in the code. Please report this to jenkinsci-users@googlegroups.com
java.lang.IllegalArgumentException: Size exceeds Integer.MAX_VALUE
    at sun.nio.ch.FileChannelImpl.map(FileChannelImpl.java:828)
    at org.owasp.dependencycheck.utils.Checksum.getChecksum(Checksum.java:66)
    at org.owasp.dependencycheck.utils.Checksum.getMD5Checksum(Checksum.java:94)
    at org.owasp.dependencycheck.dependency.Dependency.determineHashes(Dependency.java:519)
    at org.owasp.dependencycheck.dependency.Dependency.<init>(Dependency.java:109)
    at org.owasp.dependencycheck.Engine.scanFile(Engine.java:298)
    at org.owasp.dependencycheck.Engine.scanDirectory(Engine.java:277)
    at org.owasp.dependencycheck.Engine.scanDirectory(Engine.java:275)
    at org.owasp.dependencycheck.Engine.scanDirectory(Engine.java:275)
    at org.owasp.dependencycheck.Engine.scan(Engine.java:258)
    at org.owasp.dependencycheck.Engine.scan(Engine.java:201)
    at org.jenkinsci.plugins.DependencyCheck.DependencyCheckExecutor.executeDependencyCheck(DependencyCheckExecutor.java:123)
    at org.jenkinsci.plugins.DependencyCheck.DependencyCheckExecutor.performBuild(DependencyCheckExecutor.java:91)
    at org.jenkinsci.plugins.DependencyCheck.DependencyCheckBuilder$1.call(DependencyCheckBuilder.java:232)
    at org.jenkinsci.plugins.DependencyCheck.DependencyCheckBuilder$1.call(DependencyCheckBuilder.java:229)
    at hudson.remoting.LocalChannel.call(LocalChannel.java:45)
    at org.jenkinsci.plugins.DependencyCheck.DependencyCheckBuilder.perform(DependencyCheckBuilder.java:229)
    at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
    at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:772)
    at hudson.maven.MavenModuleSetBuild$MavenModuleSetBuildExecution.build(MavenModuleSetBuild.java:909)
    at hudson.maven.MavenModuleSetBuild$MavenModuleSetBuildExecution.doRun(MavenModuleSetBuild.java:676)
    at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:535)
    at hudson.model.Run.execute(Run.java:1740)
    at hudson.maven.MavenModuleSetBuild.run(MavenModuleSetBuild.java:529)
    at hudson.model.ResourceController.execute(ResourceController.java:88)
    at hudson.model.Executor.run(Executor.java:233)
```

Ideally it should just skip the file.

**Comment by eoftedal** [comment]
Wow. From sun.nio.ch.FileChannelImpl:

```
734     public MappedByteBuffer More ...map(MapMode mode, long position, long size)
735         throws IOException
736     {
...
744         if (size > Integer.MAX_VALUE)
745             throw new IllegalArgumentException("Size exceeds Integer.MAX_VALUE");
```

Why does it even take a long as argument?

**Comment by jeremylong** [comment]
Rather then skipping the file I have corrected the calculation of the digest to loop on large files to correctly generate the checksum. This has been fixed in 1.2.5-SNAPSHOT; this issue is considered closed.
